### PR TITLE
[NIFI-2812] Fix creating NiFi RPM

### PR DESCRIPTION
**Pull request by fidget83**
I've added script nifi_env.sh to bin directory and fixed jars in /lib and /lib/bootstrap directories.
Now if  I installed NiFi from rpm, its run correctly.

### CODE REVIEW

**Comment by bbende** [comment]
@fidget83 thank you for the contribution, can you please create a JIRA that describes the problem this is trying to fix? and then update this PR to include the JIRA number. Thanks.

**Comment by fidget83** [comment]
@bbende, Jira issue is https://issues.apache.org/jira/browse/NIFI-2812

**Comment by trixpan** [comment]
@fidget83 would you mind amending the commit message and adding reference to the JIRA you mentioned above?

You can do so by running `git commit --amend`

**Comment by fidget83** [comment]
@trixpan, done

File: nifi-assembly/pom.xml [hunk 1]
```diff
@@ -505,11 +510,18 @@ language governing permissions and limitations under the License. -->
                                             <directory>/opt/nifi/nifi-${project.version}/lib</directory>
                                             <dependency>
                                                 <excludes>
+                                                    <exclude>net.minidev:asm</exclude>
+                                                    <exclude>asm:asm</exclude>
```
Comment (trixpan) [inline nifi-assembly/pom.xml]: are you sure this is necessary?

File: nifi-assembly/pom.xml [hunk 2]
```diff
@@ -505,11 +510,18 @@ language governing permissions and limitations under the License. -->
                                             <directory>/opt/nifi/nifi-${project.version}/lib</directory>
                                             <dependency>
                                                 <excludes>
+                                                    <exclude>net.minidev:asm</exclude>
```
Comment (trixpan) [inline nifi-assembly/pom.xml]: are you sure this is necessary?

File: nifi-assembly/pom.xml [hunk 3]
```diff
@@ -505,11 +510,18 @@ language governing permissions and limitations under the License. -->
...
+                                                    <exclude>asm:asm</exclude>
+                                                    <exclude>org.bouncycastle:bcpkix-jdk15on</exclude>
+                                                    <exclude>com.fasterxml.jackson.core:jackson-annotations</exclude>
```
Comment (trixpan) [inline nifi-assembly/pom.xml]: seems to be absent from tar file? Sure it is required?

File: nifi-assembly/pom.xml [hunk 4]
```diff
@@ -505,11 +510,18 @@ language governing permissions and limitations under the License. -->
...
+                                                    <exclude>com.fasterxml.jackson.core:jackson-core</exclude>
```
Comment (trixpan) [inline nifi-assembly/pom.xml]: seems to be absent from tar file? Sure it is required?

File: nifi-assembly/pom.xml [hunk 5]
```diff
@@ -505,11 +510,18 @@ language governing permissions and limitations under the License. -->
...
+                                                    <exclude>com.fasterxml.jackson.core:jackson-databind</exclude>
```
Comment (trixpan) [inline nifi-assembly/pom.xml]: seems to be absent from tar file? Sure it is required?

File: nifi-assembly/pom.xml [hunk 6]
```diff
@@ -505,11 +510,18 @@ language governing permissions and limitations under the License. -->
...
+                                                    <exclude>com.jayway.jsonpath:json-path</exclude>
```
Comment (trixpan) [inline nifi-assembly/pom.xml]: seems to be absent from tar file? Sure it is required?

File: nifi-assembly/pom.xml [hunk 7]
```diff
@@ -505,11 +510,18 @@ language governing permissions and limitations under the License. -->
...
+                                                    <exclude>net.minidev:json-smart</exclude>
```
Comment (trixpan) [inline nifi-assembly/pom.xml]: seems to be absent from tar file? Sure it is required?

File: nifi-assembly/pom.xml [hunk 8]
```diff
@@ -505,11 +510,18 @@ language governing permissions and limitations under the License. -->
...
                                                     <exclude>org.apache.nifi:nifi-bootstrap</exclude>
                                                     <exclude>org.apache.nifi:nifi-resources</exclude>
                                                     <exclude>org.apache.nifi:nifi-docs</exclude>
                                                     <exclude>commons-io:commons-io</exclude>
-                                                    <exclude>org.apache.commons:commons-lang3</exclude>
```
Comment (trixpan) [inline nifi-assembly/pom.xml]: seems to be present in the tar file but no longer added to the rpm?

**Comment by fidget83** [comment]
@trixpan, I just unified the list of jars in rpm and in created .tar.gz.file. I didn't analyze necessary of each jar, but with this set nifi works :)

**Comment by trixpan** [comment]
@fidget83 the JARs should not be unified, instead only the appropriate JARs should go into each directory. Would you mind amending your commit?

Kind regards

**Comment by joaohf** [comment]
Hi,

I've tested this and the output files from rpm package is the same of tar.gz.

**Comment by trixpan** [comment]
@fidget83 - thanks for the contribution.

@joaohf - muito obrigadopelo feedback. :wink: 

Notes: created RPM. did diff between LGTM, will merge later today.

**Comment by trixpan** [comment]
Notes: 
* created RPM. 
* did diff between RPM with RPM profile and TGZ without the RPM profile
* installed from the RPM. started NiFi. Connected to it.

LGTM, will merge later today.

**Comment by joewitt** [comment]
@trixpan please make sure that commit log entries indicate/start with the JIRA.

This would be in this case

NIFI-2812 This closes #1047. Fixes rpm includes/excludes.

**Pull request closed without merging**